src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -213,10 +214,24 @@ private Regex(string pattern, RegexOptions options, TimeSpan matchTimeout, bool
                 capslist = cached._capslist;
                 capsize = cached._capsize;
                 _code = cached._code;
+                factory = cached._factory;
                 _runnerref = cached._runnerref;
                 _replref = cached._replref;
                 _refsInitialized = true;
             }
+
+            // if the compile option is set, then compile the code if it's not already
+            if (UseOptionC() && factory == null)
+            {
+                factory = Compile(_code, roptions);
+
+                if (useCache && cached != null)
+                {
+                    cached.AddCompiled(factory);
+                }
```

Comment:
Is this what netfx does?  I'm wondering, in the case of a race condition that prevents AddCompiled from actually adding it because it's already been added, whether we want to use the same one that was already added.  Maybe it doesn't matter.

Reply:
It is apparnetly what netfx does.